File: datasets/Python/Claude/Algorithm/easy1.py
def alternate(s):
    def is_alternating(string, char1, char2):
        """Check if string alternates between char1 and char2"""
        if len(string) < 2:
            return True
        
        # Filter string to contain only char1 and char2
        filtered = ''.join([c for c in string if c == char1 or c == char2])
        
        if len(filtered) < 2:
            return True
        
        # Check if it alternates
        for i in range(1, len(filtered)):
            if filtered[i] == filtered[i-1]:
                return False
        
        return True
    
    # Get all unique characters
    unique_chars = list(set(s))
    max_length = 0
    
    # Try all pairs of characters
    for i in range(len(unique_chars)):
        for j in range(i + 1, len(unique_chars)):
            char1, char2 = unique_chars[i], unique_chars[j]
            
            # Create string with only these two characters
            filtered_string = ''.join([c for c in s if c == char1 or c == char2])
            
            # Check if this filtered string alternates
            if is_alternating(filtered_string, char1, char2):
                max_length = max(max_length, len(filtered_string))
    
    return max_length

# Main function to take input
def main():
    print("=== Python Solution ===")
    
    # Take manual input
    n = int(input("Enter the length of string: "))
    s = input("Enter the string: ")
    
    result = alternate(s)
    print(f"Length of longest alternating string: {result}")

if __name__ == "__main__":
    main()

File: datasets/C++/Claude/Algorithm/easy1.cpp
#include <iostream>
#include <string>
#include <set>
#include <vector>
#include <algorithm>
using namespace std;

bool isAlternating(const string& s, char char1, char char2) {
    // Filter string to contain only char1 and char2
    string filtered;
    for (char c : s) {
        if (c == char1 || c == char2) {
            filtered += c;
        }
    }
    
    if (filtered.length() < 2) {
        return true;
    }
    
    // Check if it alternates
    for (int i = 1; i < filtered.length(); i++) {
        if (filtered[i] == filtered[i-1]) {
            return false;
        }
    }
    
    return true;
}

int alternate(const string& s) {
    // Get all unique characters
    set<char> unique_set(s.begin(), s.end());
    vector<char> unique_chars(unique_set.begin(), unique_set.end());
    
    int max_length = 0;
    
    // Try all pairs of characters
    for (int i = 0; i < unique_chars.size(); i++) {
        for (int j = i + 1; j < unique_chars.size(); j++) {
            char char1 = unique_chars[i];
            char char2 = unique_chars[j];
            
            // Create string with only these two characters
            string filtered_string;
            for (char c : s) {
                if (c == char1 || c == char2) {
                    filtered_string += c;
                }
            }
            
            // Check if this filtered string alternates
            if (isAlternating(filtered_string, char1, char2)) {
                max_length = max(max_length, (int)filtered_string.length());
            }
        }
    }
    
    return max_length;
}

int main() {
    cout << "=== C++ Solution ===" << endl;
    
    // Take manual input
    int n;
    string s;
    cout << "Enter the length of string: ";
    cin >> n;
    cout << "Enter the string: ";
    cin >> s;
    
    int result = alternate(s);
    cout << "Length of longest alternating string: " << result << endl;
    
    return 0;
}